Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm

```diff
@@ -427,37 +427,52 @@
         break;
     }
 
-    auto replaceTextIndicatorUUID = WTF::UUID::createVersion4();
-    m_webPage->createTextIndicatorForRange(replacedRange, [uuid = replaceTextIndicatorUUID, weakWebPage = WeakPtr { m_webPage }](std::optional<WebCore::TextIndicatorData>&& textIndicatorData) {
+    auto sourceTextIndicatorUUID = WTF::UUID::createVersion4();
+    auto finalTextIndicatorUUID = WTF::UUID::createVersion4();
+
+    m_webPage->createTextIndicatorForRange(replacedRange, [sourceTextIndicatorUUID = sourceTextIndicatorUUID, weakWebPage = WeakPtr { m_webPage }](std::optional<WebCore::TextIndicatorData>&& textIndicatorData) {
         if (!weakWebPage)
             return;
         RefPtr protectedWebPage = weakWebPage.get();
         if (textIndicatorData)
-            protectedWebPage->addTextIndicatorStyleForID(uuid, WebKit::TextIndicatorStyle::Source, *textIndicatorData);
+            protectedWebPage->addTextIndicatorStyleForID(sourceTextIndicatorUUID, WebKit::TextIndicatorStyle::Source, *textIndicatorData);
     });
 
     replaceContentsOfRangeInSession(uuid, resolvedRange, *fragment);
 
-    auto sessionRangeAfterReplace = contextRangeForSessionWithUUID(uuid);
-    if (!sessionRange) {
-        ASSERT_NOT_REACHED();
-        return;
-    }
     auto characterRangeAfterReplace = WebCore::CharacterRange(range.location, range.length + characterCountDelta);
 
     bool attachedToExistingSession = false;
     for (auto& session : m_textIndicatorCharacterRangesForSessions) {
```

Comment:
Nit - not new in this patch, but I think it might be a bit more readable with destructuring syntax:

```
for (auto& [sessionID, characterRanges] : m_textIndicatorCharacterRangesForSessions) {
    …
}
```